main/lsp/LSPConfiguration.cc

```diff
@@ -131,7 +131,7 @@ string LSPConfiguration::localName2Remote(string_view filePath) const {
 string LSPConfiguration::remoteName2Local(string_view uri) const {
     assertHasClientConfig();
     const bool isSorbetURI = absl::StartsWith(uri, sorbetScheme);
-    if (!absl::StartsWith(uri, clientConfig->rootUri) && !clientConfig->enableSorbetURIs && !isSorbetURI) {
```

Comment:
This is the faulty logic causing the bug.

If sorbet URIs are enabled, then this entire statement evaluates to false regardless of if the file is within the workspace.

I ended up encoding a proper version of this check in `isUriInWorkspace` and reusing it elsewhere.